spec/models/user_spec.rb

```diff
@@ -127,6 +127,58 @@
     end
   end
 
+  describe "#update" do
+    let(:existing_user) { create(:user) }
+
+    it "sets renamed_at if username changed" do
+      existing_user.update(login: "new_username")
+      expect(existing_user.renamed_at).to be_within(1.minute).of(Time.now.utc)
+    end
+
+    it "does not set renamed_at when username not changed" do
+      existing_user.update(email: "newemail@example.com")
+      expect(existing_user.renamed_at).to be_nil
+    end
+
+    context "username recently changed" do
+      before do
+        Timecop.freeze
```

Comment:
You can use [Rails' time helpers](https://api.rubyonrails.org/v5.2/classes/ActiveSupport/Testing/TimeHelpers.html#method-i-travel_to), which automatically remove the stubs at the end of specs, so you don't need an `after`.